### ISSUE TITLE
'Stored Payment Methods' section in customer dashboard incorrectly depends on Braintree being enabled

### ISSUE THREAD

**Issue by moloughlin** [magento/magento2#23205]
### Preconditions (*)
Tested against Magento Community version 2.3.1 (using latest develop branch)

### Steps to reproduce (*)
1. Implement a new payment method using the Magento_Vault saved card (token) functionality 
2. Leave Braintree disabled
3. Place an order, saving a card token against a customer (resulting in an entry in the vault_payment_token table)
4. Navigate to customer account dashboard

### Expected result (*)
1. I should be able to see and navigate to the 'Stored Payment Methods' section from my customer account dashboard

### Actual result (*)
1. Missing the 'Stored Payment Methods' section in my customer account dashboard sidebar.
![missing_stored_cards](https://user-images.githubusercontent.com/3859277/59249799-c1f22180-8c64-11e9-9055-9487f26ae6d2.png)
2. Note that I'm able to manually navigate to the vault/cards/listaction/ page and verify that there is a valid payment token stored against my customer account.

### Root Cause
app/code/Magento/Vault/view/frontend/layout/customer_account.xml implicitly depends on Braintree being enabled as a payment method. This is incorrect, since the Vault is meant as a generic framework that any payment method _should_ be able to utilize for tokenized card storage, completely independent of Braintree.
```
            <block class="Magento\Customer\Block\Account\SortLinkInterface"
                   name="customer-account-navigation-my-credit-cards-link"
                   ifconfig="payment/braintree/active"
            >
```
Note also that this introduces a circular dependency, Magento_Braintree explicitly depends on Magento_Vault (via module.xml), and this **ifconfig** node now establishes an implicit dependency in the other direction.

**Comment by Karlasa** [comment]
@moloughlin Thanks for well-formed issue report.
I agree that this exact config is not valid solution. Instead it should be something more generic setting in admin instead of braintree setting to still be able to disable link if needed. 

Cheers,
Karla

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @shikhamis11 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-17790` were created

**Issue Available**: @shikhamis11, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by ghost** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @prabhatrawat-webkul. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @prabhatrawat-webkul, here is your [Magento instance](https://i-23205-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23205-2-3-develop.instances.magento-community.engineering/admin](https://i-23205-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by svitja** [comment]
@prabhatrawat-webkul @shikhamis11 The internal issue MC-17790 - already fixed and waiting for delivery.

**Comment by magento-engcom-team** [comment]
Hi @moloughlin. Thank you for your report. 
The issue has been fixed in magento/magento2#24501 by @prabhatrawat-webkul in 2.3-develop branch
Related commit(s):
 - [1f597119af94c46d1a5d6a052393f91edca46904](https://github.com/magento/magento2/commit/1f597119af94c46d1a5d6a052393f91edca46904)

The fix will be available with the upcoming 2.3.4 release.
